build_tools/print_driver_gpu_info.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python3
+"""
+Sanity check script for CI runners.
+
+Prints a small set of driver / GPU information from:
+  - amd-smi
+  - rocminfo
+
+It prefers binaries installed under ./build/bin (as produced by
+install_rocm_from_artifacts.py), and falls back to the PATH.
+
+Usage:
+    ./build_tools/print_driver_gpu_info.py
+"""
+
+import argparse
+import json
+import logging
+import re
+import subprocess
+import sys
+from typing import Dict, Tuple, Optional, List, Any
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+log = logging.info
+
+
+def run_cmd(cmd: list[str]) -> Tuple[bool, str]:
+    """
+    Run a command and return (ok, combined_output).
+
+    ok = True  -> command executed and returned success (exit code 0)
+    ok = False -> command missing or returned a non-zero exit code
+    """
+    try:
```

Comment:
Let's use the same style of helper function from our other scripts: https://github.com/ROCm/TheRock/blob/d76278526218def9fb1b016bc9e421738cb4f8f6/build_tools/fetch_sources.py#L39-L43